File: benchmarks/accuracy/fixtures/java/control_parameterized_sql.java
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import javax.servlet.http.HttpServletRequest;

public class UserLookup {
    public ResultSet find(HttpServletRequest request, Connection conn) throws SQLException {
        String name = request.getParameter("name");
        PreparedStatement ps = conn.prepareStatement("SELECT id, email FROM users WHERE name = ?");
        ps.setString(1, name);
        return ps.executeQuery();
    }
}

File: benchmarks/accuracy/fixtures/python/control_parameterized_sql.py
from flask import request
import sqlite3


def find_user():
    conn = sqlite3.connect("app.db")
    cursor = conn.cursor()
    username = request.args.get("username")
    query = "SELECT id, email FROM users WHERE username = ?"
    cursor.execute(query, (username,))
    return cursor.fetchone()
